### PR TITLE
feat: initial scaffolding for directive validation and git-based directives

### PR DESCRIPTION
**Pull request by krancour**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -151,20 +155,21 @@ codegen-controller:
...
 .PHONY: codegen-ui
 codegen-ui:
 	pnpm --dir=ui install --dev
 	pnpm --dir=ui run generate:schema
 
+.PHONY: codegen-docs
+codegen-docs:
+	npm install -g @bitnami/readme-generator-for-helm
+	bash hack/helm-docs/helm-docs.sh
+
```
Comment (krancour) [inline Makefile:187]: No changes here. I just re-ordered the targets so they are defined in the same order they are executed when using `make codegen`. I thought it made it slightly easier to follow.

File: hack/codegen/directive-configs.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  internal/directives/schemas/*.json
+
+printf "${generated_code_warning}$(cat ${out_file})" > ${out_file}
+
+# Pointers to bools and strings don't make a lot of sense in most cases.
+#
+# Note that -i works on Linux, but not on macOS. -i '' works on macOS, but not
+# on Linux. So we use -i.bak, which works on both.
+sed -i.bak 's/\*bool/bool/g' ${out_file}
+sed -i.bak 's/\*string/string/g' ${out_file}
```
Comment (krancour) [inline hack/codegen/directive-configs.sh:23]: The tool uses pointers for anything optional, which is annoying AF for strings and bools (maybe ints, too). There wasn't a flag to change that behavior, so I resorted to this.

File: internal/directives/copy_directive.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import (
 	"fmt"
 )
 
-func init()  {
+func init() {
```
Comment (krancour) [inline internal/directives/copy_directive.go:9]: Linter caught this.

File: internal/directives/directive.go [hunk 4]
```diff
@@ -92,4 +92,3 @@ func configToStruct[T any](c Config) (T, error) {
 
 	return result, nil
 }
-
```
Comment (krancour) [inline internal/directives/directive.go:95]: Linter caught this.

File: internal/directives/registry.go [hunk 5]
```diff
@@ -17,7 +17,6 @@ func BuiltinsRegistry() DirectiveRegistry {
 // register and retrieve directives by name.
 type DirectiveRegistry map[string]Directive
 
-
```
Comment (krancour) [inline internal/directives/registry.go:20]: Linter caught this.

File: internal/directives/engine.go [hunk 6]
```diff
@@ -47,11 +47,11 @@ func NewEngine(
 // Execute runs the provided list of directives in sequence.
 func (e *Engine) Execute(ctx context.Context, steps []Step) (Result, error) {
 	// TODO(hidde): allow the workDir to be restored from a previous execution.
-	workDir, err := os.CreateTemp("", "run-")
+	workDir, err := os.MkdirTemp("", "run-")
```
Comment (hiddeco) [inline internal/directives/engine.go:50]: 🤦 doh.

**Review by hiddeco — APPROVED** [review]
We already talked through the details of this offline, so lets just get it merged. 💯

**Pull request merged**